### PR TITLE
Update to latest azure-metrics-exporter to remove the go.mod replace

### LINKED ISSUE #3424: Get changes of `azure-metrics-exporter` merged upstream and remove `replace` in go.mod

### PR DESCRIPTION
**Pull request by kgeckhart**
#### PR Description
Updates to the latest version of https://github.com/webdevops/azure-metrics-exporter to drop the go.mod replace. Also included some upstream changes to move to zap logging and adjust how resource tags are handled. Tested with previous configs used

#### Which issue(s) this PR fixes

Fixes #3424

#### PR Checklist

- [x] CHANGELOG updated
- [n/a] Documentation added
- [x] Tests updated

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -51,6 +51,8 @@ Main (unreleased)
 
 - Update OracleDB Exporter dependency to 0.5.0 (@schmikei)
 
+- Update azure-metrics-exporter to latest (@kgeckhart)
```
Comment (rfratto) [inline CHANGELOG.md:54]: Can you put an explicit version here? "latest" will change with time.

Comment (kgeckhart) [reply CHANGELOG.md]: Updated, unfortunately the upstream author hasn't done a release since Sept 2022

File: go.mod [hunk 2]
```diff
@@ -600,6 +600,8 @@ require (
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
 )
 
+require github.com/prometheus/tsdb v0.7.1
```
Comment (jcreixell) [inline go.mod:603]: is this required? (seems to work for me without it)

**Review by jcreixell — APPROVED** [review]
LGTM

**Pull request merged**